addons/mail/models/mail_thread.py

```diff
@@ -601,27 +601,40 @@ def _message_track(self, tracked_fields, initial):
         and initial values, return a structure that is a tuple containing :
 
          - a set of updated column names
-         - a list of changes (initial value, new value, column name, column info) """
+         - a list of ORM (0, 0, values) commands to create 'mail.tracking.value' """
         self.ensure_one()
         changes = set()  # contains onchange tracked fields that changed
         tracking_value_ids = []
 
-        # generate tracked_values data structure: {'col_name': {col_info, new_value, old_value}}
+        # Check which fields have changed
+        modified_tracked_fields = {}
         for col_name, col_info in tracked_fields.items():
             initial_value = initial[col_name]
             new_value = getattr(self, col_name)
-
             if new_value != initial_value and (new_value or initial_value):  # because browse null != False
-                tracking_sequence = getattr(self._fields[col_name], 'tracking',
-                                            getattr(self._fields[col_name], 'track_sequence', 100))  # backward compatibility with old parameter name
-                if tracking_sequence is True:
-                    tracking_sequence = 100
-                tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info, tracking_sequence)
-                if tracking:
-                    tracking_value_ids.append([0, 0, tracking])
-
-                if col_name in tracked_fields:
-                    changes.add(col_name)
+                modified_tracked_fields[col_name] = dict(col_info, new_value=new_value, initial_value=initial_value)
+
+        if modified_tracked_fields:
+            # load `ir.model.fields` records associated and add them to the info dict. Use sudo to avoid more queries
+            ir_fields = self.env['ir.model.fields'].sudo().search([('name', 'in', tuple(modified_tracked_fields.keys())), ('model', '=', self._name)])
+            modified_tracked_fields = {
+                field.name: dict(modified_tracked_fields.get(field.name), ir_field=field)
```

Comment:
If for any reasons `fields.name` is not in `modified_tracked_fields` the method `get` will return `None` which will trigger a `TypeError`
the code should probably look like that
``` python
modified_tracked_fields = {
field.name: dict(modified_tracked_fields.get(field.name, {}), ir_field=field)
for field in ir_fields
}
```